foreshadow/tests/test_foreshadow.py

```diff
@@ -746,6 +751,71 @@ def test_foreshadow_serialization_adults_small_classification():
     assertions.assertAlmostEqual(score1, score2, places=7)
 
 
+def test_foreshadow_serialization_adults_small_classification_override():
+    from foreshadow.foreshadow import Foreshadow
+    import pandas as pd
+    import numpy as np
+    from sklearn.model_selection import train_test_split
+    from sklearn.linear_model import LogisticRegression
+
+    np.random.seed(1337)
+
+    adult = pd.read_csv("examples/adult_small.csv")
+    X_df = adult.loc[:, "age":"workclass"]
+    y_df = adult.loc[:, "class"]
+
+    X_train, X_test, y_train, y_test = train_test_split(
+        X_df, y_df, test_size=0.2
+    )
+
+    shadow = Foreshadow(estimator=LogisticRegression())
+    shadow.fit(X_train, y_train)
+    shadow.to_json("foreshadow_adults_small_logistic_regression_1.json")
+    score1 = shadow.score(X_test, y_test)
+
+    from foreshadow.intents import IntentType
+
+    shadow.override_intent("age", IntentType.NUMERIC)
+    shadow.override_intent("workclass", IntentType.CATEGORICAL)
+    shadow.fit(X_train, y_train)
+    shadow.to_json("foreshadow_adults_small_logistic_regression_2.json")
```

Comment:
Shall we check the intent overrides are actually respected?

Reply:
I inspected the json file and the override section is present in the cache manager but yes we could add an extra unit test to check if intent override is present.